src/system_info/system_info_network_mobile.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2013 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "system_info/system_info_network.h"
+
+SysInfoNetwork::SysInfoNetwork()
+    : type_(SYSTEM_INFO_NETWORK_UNKNOWN) {}
+
+SysInfoNetwork::~SysInfoNetwork() {}
+
+void SysInfoNetwork::StartListening() {
+  vconf_notify_key_changed(VCONFKEY_NETWORK_STATUS,
+      (vconf_callback_fn)OnNetworkTypeChanged, this);
+
+  vconf_notify_key_changed(VCONFKEY_TELEPHONY_SVCTYPE,
+      (vconf_callback_fn)OnNetworkTypeChanged, this);
+}
+
+void SysInfoNetwork::StopListening() {
+  vconf_ignore_key_changed(VCONFKEY_NETWORK_STATUS,
+      (vconf_callback_fn)OnNetworkTypeChanged);
+
+  vconf_ignore_key_changed(VCONFKEY_TELEPHONY_SVCTYPE,
+      (vconf_callback_fn)OnNetworkTypeChanged);
+}
+
+void SysInfoNetwork::GetCellularNetworkType() {
+  int service_type = 0;
+  if (vconf_get_int(VCONFKEY_TELEPHONY_SVCTYPE, &service_type))
+    return;
+
+  switch (service_type) {
+    case VCONFKEY_TELEPHONY_SVCTYPE_NONE:
+    case VCONFKEY_TELEPHONY_SVCTYPE_NOSVC:
+    case VCONFKEY_TELEPHONY_SVCTYPE_EMERGENCY:
+      type_ = SYSTEM_INFO_NETWORK_NONE;
+      break;
+    case VCONFKEY_TELEPHONY_SVCTYPE_2G:
+      type_ = SYSTEM_INFO_NETWORK_2G;
+      break;
+    case VCONFKEY_TELEPHONY_SVCTYPE_2_5G:
+    case VCONFKEY_TELEPHONY_SVCTYPE_2_5G_EDGE:
+      type_ = SYSTEM_INFO_NETWORK_2_5G;
+      break;
+    case VCONFKEY_TELEPHONY_SVCTYPE_3G:
+      type_ = SYSTEM_INFO_NETWORK_3G;
+      break;
+    case VCONFKEY_TELEPHONY_SVCTYPE_HSDPA:
+      type_ = SYSTEM_INFO_NETWORK_4G;
+      break;
+    default:
+      type_ = SYSTEM_INFO_NETWORK_UNKNOWN;
+  }
+}
+
+void SysInfoNetwork::OnNetworkTypeChanged(keynode_t* node, void* user_data) {
```

Comment:
The function body is same with <code>OnNetworkTypeChanged</code> in <code>system_info_network_ivi.cc</code>. Please put it in <code>system_info_network.cc</code>.